011-kafka-canary.md

```diff
@@ -0,0 +1,40 @@
+# Strmixi Canary
+
+Implement a Kafka canary tool which will act as an indicator of whether Kafka clusters are operating correctly. This would be achieved by creating a canary topic and continously producing and consuming events on the topics. 
+
+## Current situation
+
+Currently  Strimzi does not have any canary style feature. It does have readiness checks for Kafka cluster which can be replaced/augmented with the canary.
+
+## Motivation
+
+The canary would provide the following:
+  
+  * Improved readiness check for Kafka clusters
+  * SLI(s) for service admins
+  * Metrics:
+    * End-to-end latency
```

Comment:
If this can also be subdivided to also expose producer & consumer metrics, similar to what's shown in https://www.datadoghq.com/blog/monitoring-kafka-performance-metrics/#kafka-producer-metrics, it would help with deeper troubleshooting of where problems could be coming from

Reply:
This is the way we use for our HTTP Kafka bridge component because it internally uses native Kafka producer and consumer for bridging the HTTP traffic. We have a JMX collector and then a metrics reporter to provide these metrics in Prometheus format. I think it should work in the same way.
You can also refer to the official Kafka documentation:

https://kafka.apache.org/documentation/#producer_monitoring

https://kafka.apache.org/documentation/#consumer_monitoring

Reply:
@ppatierno you're right we can get metrics from the producer and consumer themselves, but this would give us a little more insight. For example knowing the producer latency it useful; `request-latency-avg` isn't quite the same thing since it includes non-produce requests and it's also only an average. Measuring directly could yield a histogram.